src/runtime-rs/crates/resource/src/share_fs/share_virtio_fs.rs

```diff
@@ -75,34 +77,39 @@ pub(crate) async fn setup_inline_virtiofs(id: &str, h: &dyn Hypervisor) -> Resul
     let rw_source = utils::get_host_rw_shared_path(id).join(PASSTHROUGH_FS_DIR);
     utils::ensure_dir_exist(&rw_source).context("ensure directory exist")?;
 
-    let ro_source = utils::get_host_ro_shared_path(id).join(PASSTHROUGH_FS_DIR);
-    let source = String::from(ro_source.to_str().unwrap());
-
-    let virtio_fs = ShareFsMountDevice {
-        config: ShareFsMountConfig {
-            source: source.clone(),
-            fstype: ShareFsMountType::PASSTHROUGH,
-            mount_point: mnt,
-            config: None,
-            tag: String::from(MOUNT_GUEST_TAG),
-            op: ShareFsOperation::Mount,
-            prefetch_list_path: None,
-        },
+    let host_ro_shared_path = utils::get_host_ro_shared_path(id);
+    let source = host_ro_shared_path
+        .join(PASSTHROUGH_FS_DIR)
+        .display()
+        .to_string();
+
+    let virtiofs_mount = ShareFsMountConfig {
+        source: source.clone(),
+        fstype: ShareFsMountType::PASSTHROUGH,
+        mount_point: mnt,
+        config: None,
+        tag: String::from(MOUNT_GUEST_TAG),
+        op: ShareFsOperation::Mount,
+        prefetch_list_path: None,
+    };
+
+    let sharefs_config = ShareFsConfig {
+        shared_path: host_ro_shared_path.display().to_string(),
+        mount_config: Some(virtiofs_mount),
+        ..Default::default()
     };
 
-    let result = h
-        .add_device(DeviceType::ShareFsMount(virtio_fs))
+    // update virtio-fs device with ShareFsMountConfig
+    do_update_device(d, &DeviceConfig::ShareFsCfg(sharefs_config.clone()))
```

Comment:
Do we need to `clone()` `sharefs_config` here? I don't see it referenced anymore after this point.

Reply:
Fixed